### DISCUSSION TITLE
A bug in Propeller gyroscopic torque formula?

### DISCUSSION THREAD

**Discussion by msurov** [JSBSim-Team/jsbsim#493] [Q&A]
I am looking at the code computing the gyroscopic torque at [FGPropeller.cpp:284-295](https://github.com/JSBSim-Team/jsbsim/blob/master/src/models/propulsion/FGPropeller.cpp#L284):
```c++
FGColumnVector3 vH(Ixx*omega*Sense*Sense_multiplier, 0.0, 0.0);
...
vMn = in.PQRi*(mT*vH) + mT*vTorque;
```
and a bit confused. The reason is it's missing the coefficient 0.5 in `Ixx*omega*Sense*Sense_multiplier` term. Actually, the gyroscopic torque depends on the differences `Ixx-Iyy` and `Ixx-Izz`. According to the [documentation ](https://wiki.flightgear.org/JSBSim_Thrusters#Parameter_definitions_2) the values `Iyy,Izz` are not supported. Nevertheless, If the propeller is thought as 4-blades or just a disk, then `Iyy = Izz = 0.5 * Ixx`. If the propeller is 2 blades, then vibrations will occur, but the average value of the torque also needs coefficient `0.5`.
Also, the sign of the resulting torque seems incorrect (upt: sorry, this bug was already fixed).

Where this formula is taken from?

**Comment by bcoconni** [comment]
Hi Maksim,

First,sorry for the **very** long delay before answering your message.

I am not sure where the coefficient `0.5` you are mentioning is coming from. In the code line you are referring to, `vH` is a torque which is equal to the inertia `Ixx` times the rotation speed `omega`. There is no coefficient `0.5` involved in the formula `I*ω` as far as I know.
> Actually, the gyroscopic torque depends on the differences `Ixx-Iyy` and `Ixx-Izz`.

The gyroscopic torque that is computed by JSBSim is not the gyroscopic torque of the propeller upon itself but instead the gyroscopic torque that the propeller applies to the whole aircraft. In the file `FGPropeller.cpp`, the variable `Ixx` is the propeller inertia about its rotation axis but the resisting inertia are the aircraft inertia which are computed by `FGMassBalance.cpp`.
> Nevertheless, If the propeller is thought as 4-blades or just a disk, then `Iyy = Izz = 0.5 * Ixx`.

Since the propeller is rotating in its local frame axis `X`, there is little point in providing the values `Iyy` and `Izz` as they are supposed to be included in the aircraft global inertia tensor.

**Comment by msurov** [comment]
The expression `vH = Ixx*omega` is the angular momentum, not torque. The gyroscopic torque in general is `$\omega \times J \omega$`. The cross product will give you the coefficients `Ixx - Iyy`, `Ixx - Iyy`. 

Actually, I can formulate a simple question: where did you take this term
```in.PQRi*(mT*vH)```
from? Is there anything in documentation? References to papers, etc?

**Reply by bcoconni** [reply]
> Actually, I can formulate a simple question: where did you take this term
```in.PQRi*(mT*vH)```
from? Is there anything in documentation? References to papers, etc?

This formula can be found in Aircraft Control and Simulation - 3rd edition by Stevens, Lewis and Johnson more specifically formulas 8.2-28 and 8.2-29.

**Comment by seanmcleod** [comment]
@msurov here is the relevant snippet in case you don't have access to Aircraft Control and Simulation - 3rd edition by Stevens, Lewis and Johnson.

![image](https://user-images.githubusercontent.com/824856/136958481-6f7e2e97-3adb-4dd0-8236-632bcb5087c2.png)
